test/metric_value_benchmark/base_test.go

```diff
@@ -65,23 +67,29 @@ func (t *TestRunner) runAgent() (status.TestGroupResult, error) {
 
 	agentConfigPath := filepath.Join(agentConfigDirectory, t.testRunner.getAgentConfigFileName())
 	log.Printf("Starting agent using agent config file %s", agentConfigPath)
-	test.CopyFile(agentConfigPath, configOutputPath)
-	err := test.StartAgent(configOutputPath, false)
+	common.CopyFile(agentConfigPath, configOutputPath)
+	err := common.StartAgent(configOutputPath, false)
 
 	if err != nil {
 		testGroupResult.TestResults[0].Status = status.FAILED
-		return testGroupResult, fmt.Errorf("Agent could not start due to: %v", err.Error())
+		return testGroupResult, fmt.Errorf("Agent could not start due to: %s", err.Error())
+	}
+
+	err = t.testRunner.setupAfterAgentRun()
```

Comment:
When we introduce this, we increase the time it takes before we sleep. I don't think that will cause a big issue, but it's something that we have to keep in mind. There are a lot of tests now, they each take some time, and they all run sequentially.

Reply:
True. That's why I have some ideas of improving it and align with Chad's test plan. But I guess its not worth to do here